include/tuvx/linalg/linalg.inl

```diff
@@ -0,0 +1,83 @@
+
+#include <vector>
+namespace tuvx {
+namespace linalg {
+
+template <typename T>
+inline std::vector<T> dot(trid_mat<T> A, std::vector<T> x) {
+  int size = x.size();
```

Comment:
```suggestion
  std::size_t size = x.size();
```
or use `auto` for the type. In either case, this reduces compiler warnings about losing precision when we use these sizes in other places